lib/iris/tests/unit/fileformats/grib/test_GribWrapper.py

```diff
@@ -88,90 +88,54 @@ def test_edition_1(self):
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, _message_length * i)
-
-    def test_regular_mixed(self):
+        self.patch('iris.fileformats.grib.GribWrapper._confirm_in_scope')
+        self.patch('iris.fileformats.grib.GribWrapper._compute_extra_keys')
+        self.patch('gribapi.grib_get_long', _mock_grib_get_long)
+        self.patch('gribapi.grib_get_string', _mock_grib_get_string)
+        self.patch('gribapi.grib_get_native_type', _mock_grib_get_native_type)
         tell_tale = np.arange(1, 5) * _message_length
-        expected = tell_tale - _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+        self.expected = tell_tale - _message_length
+        self.grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+        self.dtype = np.float64
+        self.fill_value = np.nan
+        self.path = self.grib_fh.name
+        self.lookup = _mock_grib_get_long
+
+    def test_regular_proxy_args(self):
         grib_message = 'regular_ll'
-        for offset in expected:
-            gw = GribWrapper(grib_message, grib_fh)
-            self.assertTrue(is_lazy_data(gw._data))
-            proxy = as_concrete_data(gw._data)
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (10, 20))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, offset)
-
-    def test_reduced_sequential(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
-        grib_message = 'reduced_gg'
-        for i, _ in enumerate(tell_tale):
-            gw = GribWrapper(grib_message, grib_fh)
-            self.assertTrue(is_lazy_data(gw._data))
-            proxy = as_concrete_data(gw._data)
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (200,))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, _message_length * i)
-
-    def test_reduced_mixed(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        expected = tell_tale - _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+        shape = (self.lookup(grib_message, 'Nj'),
+                 self.lookup(grib_message, 'Ni'))
+        for offset in self.expected:
+            with mock.patch('iris.fileformats.grib.GribDataProxy') as mock_gdp:
+                gw = GribWrapper(grib_message, self.grib_fh)
+            mock_gdp.assert_called_once_with(shape, self.dtype, self.fill_value,
+                                         self.path, offset)
```

Comment:
@corinnebosley Nice. Just need to fix the pep8 failures on the above two lines ...